tests/e2e_operations.py

```diff
@@ -750,6 +787,50 @@ def run_cose_signatures_config_check(args):
                     ), f"Failed to get receipt for txid {txid} after {max_retries} retries"
 
 
+def run_empty_ledger_dir_check(args):
+    with infra.network.network(
+        args.nodes,
+        args.binary_dir,
+        args.debug_nodes,
+        args.perf_nodes,
+        pdb=args.pdb,
+    ) as network:
+        LOG.info("Check that empty ledger directory is handled correctly")
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            LOG.debug(f"Using {tmp_dir} as ledger directory")
+
+            dir_name = os.path.basename(tmp_dir)
+
+            # Check tmp_dir is indeed empty
+            assert len(os.listdir(tmp_dir)) == 0, tmp_dir
+
+            # Start network, this should not fail
+            network.start_and_open(args, ledger_dir=tmp_dir)
+            primary, _ = network.find_primary()
+            network.stop_all_nodes()
+
+            # Now write a file in the directory
+            with open(os.path.join(tmp_dir, "ledger_1000_1500.committed"), "wb") as f:
+                f.write(b"bar")
+
+            # Start new network, this should fail
+            try:
+                network.start(args, ledger_dir=tmp_dir)
+            except:
+                pass
+
+            # Check that the node has failed with the expected error message
```

Comment:
There is a check_log_for_error_message() that essentially does that: https://github.com/microsoft/CCF/blob/ddd541a7bab23c7b7ea2005424be8257a990d3f3/tests/recovery.py#L275